src/enzo/ActiveParticle_SmartStar.C

```diff
@@ -0,0 +1,1058 @@
+	np->RadiationLifetime=SmartStarSMSLifetime*3.154e7/data.TimeUnits;
+	np->NotEjectedMass = 0.0;
+	density[index] = DensityThreshold;
+	//printf("%s: Particle Created!\n", __FUNCTION__);
+	//printf("%s: Total Mass in Accretion Region = %g Msolar (Threshold = %g)\n", __FUNCTION__,
+	//     TotalMass*ConverttoSolar, (double)MASSTHRESHOLD);
+      } // i
+    } // j
+  } // k
+  if(data.NumberOfNewParticles > 0) {
+    printf("Particles (%d) Created and done in Evaulate Formation\n", data.NumberOfNewParticles);
+    fflush(stdout);
+  }
+
+#if CALCDIRECTPOTENTIAL
+  delete [] PotentialField;
+#endif
+  return SUCCESS;
+}
+
+/*
+ * This routine handles the stellar feedback including the 
+ * accretion luminosity and supernova feedback. 
+ * The jet feedback and thermal feedback are
+ * calculated in SmartStarParticleFeedback and called from AfterEvolveLevel
+ * The radiative feedback is called from BeforeEvolveLevel
+ */
+int ActiveParticleType_SmartStar::EvaluateFeedback(grid *thisgrid_orig,
+						   ActiveParticleFormationData &data)
+{
+  SmartStarGrid *thisGrid =
+    static_cast<SmartStarGrid *>(thisgrid_orig);
+ 
+  float *density = thisGrid->BaryonField[data.DensNum];
+  float *totalenergy = thisGrid->BaryonField[data.TENum];
+  float *gasenergy = thisGrid->BaryonField[data.GENum];
+  FLOAT Time = thisGrid->ReturnTime();
+  float dt = thisGrid->dtFixed;
+  float DensityUnits, LengthUnits, TemperatureUnits, TimeUnits,
+    VelocityUnits;
+  double MassUnits;
+  GetUnits(&DensityUnits, &LengthUnits, &TemperatureUnits,
+	   &TimeUnits, &VelocityUnits, Time);
+  int npart = thisGrid->NumberOfActiveParticles;
+  FLOAT dx = thisGrid->CellWidth[0][0];
+  const double LumConvert =   POW(LengthUnits,2)/POW(TimeUnits,3);
+  MassUnits = DensityUnits * POW(LengthUnits,3);
+  double MassConversion =  ((double)dx*dx*dx * MassUnits);  //convert to g
+  /* 
+   * This routine needs to be called here. Do not return before 
+   * this point!
+   */
+  for (int n = 0; n < npart; n++) {
+    ActiveParticleType_SmartStar *ThisParticle =
+      static_cast<ActiveParticleType_SmartStar*>(
+              thisGrid->ActiveParticles[n]);
+
+    ThisParticle->level = data.level;
+  }
+
+  if(SmartStarFeedback == FALSE)
+    return SUCCESS;
+  for (int n = 0; n < npart; n++) {
+    ActiveParticleType_SmartStar *ThisParticle =
+      static_cast<ActiveParticleType_SmartStar*>(
+              thisGrid->ActiveParticles[n]);
+    
+    if(BH == ThisParticle->ParticleClass)
+      continue;
+    /* The first form of feedback is accretion luminosity 
+     * This is given by L = G M_acc M_* / R_*
+     */
+    /* StarMass in units of Msolar */
+    float StarMass = ThisParticle->Mass * MassConversion / SolarMass;
+    /* Accretion Rate in units of Msolar/s */
+    float AccretionRate = ThisParticle->AccretionRate[ThisParticle->TimeIndex]*MassUnits/(SolarMass*TimeUnits);
+    if(AccretionRate*3.154e7 < 1e-30) /* Takes care of any negative accretion rates */
```

Comment:
Replace hard-coded 3.154e7 throughout the file with yr_s from phys_const.def

Reply:
Done